lib/archethic/transaction_chain.ex

```diff
@@ -262,6 +266,37 @@ defmodule Archethic.TransactionChain do
     get_transaction(address, fields)
   end
 
+  @doc """
+  Get the first transaction from a given chain address
+  """
+  @spec get_first_transaction(binary(), list()) ::
+          {:ok, Transaction.t()} | {:error, :transaction_not_exists}
+  def get_first_transaction(address, fields \\ []) when is_binary(address) do
+    case get_first_transaction_address(address) do
+      {:ok, address} -> get_transaction(address, fields)
+      {:error, :transaction_not_exists} -> {:error, :transaction_not_exists}
+    end
+  end
+
+  @doc """
+  Get the first transaction Address from a genesis/chain address
+  """
+  @spec get_first_transaction_address(address :: binary()) ::
+          {:ok, address :: binary()} | {:error, :transaction_not_exists}
+  def get_first_transaction_address(address) when is_binary(address) do
+    address_list =
+      address
+      |> get_genesis_address()
+      |> list_chain_addresses()
+      |> Enum.at(0)
+
+    case address_list do
+      nil -> {:error, :transaction_not_exists}
+      [] -> {:error, :transaction_not_exists}
```

Comment:
This cas will never happen as `Enum.at(0)` will return `nil` or `{address, timestamp}`